source/exe/main.cc

```diff
@@ -33,10 +35,36 @@ class ProdComponentFactory : public ComponentFactory {
 
 } // Server
 
+namespace {
+
+int validateConfig(OptionsImpl& options, Server::ProdComponentFactory& component_factory,
+                   const LocalInfo::LocalInfoImpl& local_info) {
+  Server::ValidationHotRestart restarter;
+  Logger::Registry::initialize(options.logLevel(), restarter.logLock());
+  Stats::HeapRawStatDataAllocator alloc;
+  Stats::ThreadLocalStoreImpl stats_store(alloc);
+
+  Server::ValidationInstance server(options, restarter, stats_store, restarter.accessLogLock(),
+                                    component_factory, local_info);
+  std::cout << "configuration '" << options.configPath() << "' OK" << std::endl;
+  server.shutdown();
+  return 0;
+}
+
+} // namespace
+
 int main(int argc, char** argv) {
-  ares_library_init(ARES_LIB_INIT_ALL);
   Event::Libevent::Global::initialize();
   OptionsImpl options(argc, argv, Server::SharedMemory::version(), spdlog::level::warn);
+  Server::ProdComponentFactory component_factory;
+  LocalInfo::LocalInfoImpl local_info(Network::Utility::getLocalAddress(), options.serviceZone(),
+                                      options.serviceClusterName(), options.serviceNodeName());
+
+  if (options.mode() != Server::Mode::Serve) {
+    return validateConfig(options, component_factory, local_info);
```

Comment:
Do you want to add a `if (option.mode() == Server::Mode::Validate) { NOT_IMPLEMENTED; }`? That would avoid someone asking for light mode and getting back full validation behavior.